### PR TITLE
Mech/Tag for SeaPickles' pickle_count

### PR DESCRIPTION
**Pull request by Ph4i1ur3**
As requested by a Community member.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java [hunk 1]
```diff
@@ -146,6 +146,8 @@ public static void registermainProperties() {
         // register core MaterialTag properties
         if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_13)) {
             PropertyParser.registerProperty(MaterialAge.class, MaterialTag.class);
+            PropertyParser.registerProperty(MaterialPickle.class, MaterialTag.class);
+            PropertyParser.registerProperty(MaterialRepeater.class, MaterialTag.class);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: ^ Started a second PR before finishing the first and didn't hide all the evidence, eh?

Comment (Ph4i1ur3) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: OOF, lol

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public SeaPickle getSeaPickle() {
+        return (SeaPickle) material.getModernData().data;
+    }
+
+    public int getCurrent() {
+        return getSeaPickle().getPickles();
+    }
+
+    public int getMax() {
+        return getSeaPickle().getMinimumPickles();
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java]: ^ Nope. Copypasta makes spaghetti and all that.

Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java]: These min/max values should have tags btw.

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        // @input ElementTag(Number)
+        // @description
+        // Sets the amount of pickles in a Sea Pickle material.
+        // @tags
+        // <MaterialTag.pickle_count>
+        // -->
+        if (mechanism.matches("pickle_count") && mechanism.requireInteger()) {
+            int count = mechanism.getValue().asInt();
+            if (count < getMin() || count > getMax()) {
+                Debug.echoError("Pickle count value '" + count + "' is not valid. Must be between" + getMin() + " and " + getMax() + ".");
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java]: `Must be between" + getMin() +` <-- missing a space there.

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        material = _material;
+    }
+
+    MaterialTag material;
+
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.pickle_count>
+        // @returns ElementTag(Number)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialPickle.java:44]: This meta needs an `@mechanism` link.

**Pull request merged**